software/main.py

```diff
@@ -138,52 +179,157 @@ def get_image_data(self):
 
         if self.prev_ball_count != self.ball_count:
             # this is for debugging
-            #print("ball_count: {}".format(self.ball_count))
+            #self.logger.log.info("ball_count: {}".format(self.ball_count))
             self.prev_ball_count = self.ball_count
 
         # first element is always the closest ball
         if self.ball_count > 0:
             self.ball = self.processed_data.balls[0]
+            self.last_seen_ball = self.ball
         elif self.no_balls_frames >= self.max_ball_miss:  # use old ball sometimes
             self.ball = None
 
-        if self.basket_color == Color.MAGENTA:
-            self.basket = self.processed_data.basket_m
-        elif self.basket_color == Color.BLUE:
-            self.basket = self.processed_data.basket_b
+        # TODO - implement code to prevent our robot from driving over lines
+        self.baskets[Color.MAGENTA] = self.processed_data.basket_m
+        self.baskets[Color.BLUE] = self.processed_data.basket_b
+
+        if self.current_state not in [State.Stopped, State.RemoteControl, State.Debug]:
+            if self.search_substate == SearchState.DriveToSearch:
+                return
+            for basket in self.baskets:
+                if self.baskets[basket].exists:
+                    if self.baskets[basket].distance < 600:
```

Comment:
Similar values used in image processor as well. Maybe define some distance rules in a separate module and use them everywhere where it is needed

Reply:
Made it a variable in this instance. Haven't touched the image processor one yet, because I'm still deciding what to do with it.